test/Spec.hs

```diff
@@ -150,7 +152,65 @@ processTicketSpec =
                         zendeskResponses <- run appExecution
 
                         -- Check we have some comments.
-                        assert $ length zendeskResponses > 0
+                        assert $ not (null zendeskResponses)
+
+        it "processes ticket, with no attachments" $
+            forAll (listOf1 genCommentWithNoAttachment) $ \(commentsWithoutAttachment :: [Comment]) ->
+                forAll arbitrary $ \ticketInfo ->
+                    monadicIO $ do
+
+                    let stubbedZendeskLayer :: ZendeskLayer App
+                        stubbedZendeskLayer =
+                            emptyZendeskLayer
+                                { zlGetTicketComments = \_ -> pure commentsWithoutAttachment
+                                , zlGetTicketInfo     = \_ -> pure ticketInfo
+                                , zlPostTicketComment = \_ -> pure ()
+                                }
+
+                    let stubbedConfig :: Config
+                        stubbedConfig = withStubbedIOAndZendeskLayer stubbedZendeskLayer
+
+                    let appExecution :: IO [ZendeskResponse]
+                        appExecution = runApp (processTicket . ticketId $ ticketInfo) stubbedConfig
+
+                    zendeskResponses <- run appExecution
+
+                    assert $ not (null zendeskResponses) && isTaggedWithNoLogs zendeskResponses
+
+        it "processes ticket, with attachments" $
+            forAll (listOf1 arbitrary) $ \(comments:: [Comment]) ->
+                forAll arbitrary $ \ticketInfo ->
+                    monadicIO $ do
+
+                    pre $ any (not . null . cAttachments) comments
```

Comment:
`all (not . null . cAttachments) comments` - we require that all the comments have attachments, right?

Reply:
Yes, good thinking but I didn't even notice we were returning multiple `ZendeskResponse`, my bad.